tribe-ext-divi-theme-compatibility.php

```diff
@@ -0,0 +1,272 @@
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Example' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Divi_Theme_Compatibility extends Tribe__Extension {
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-divi-theme-compatibility', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			/**
+			 * All extensions require PHP 5.6+, following along with https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/
+			 */
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
+					$message = '<p>';
+					$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', 'tribe-ext-divi-theme-compatibility' ), $this->get_name(), $php_required_version );
+					$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+					$message .= '</p>';
+					tribe_notice( $this->get_name(), $message, 'type=error' );
+				}
+
+				return;
+			}
+
+			if ( function_exists( 'et_custom_posts_per_page' ) && ! is_admin() ) {
+				add_filter( 'parse_query', array( $this, 'remove_et_custom_posts_per_page' ), 100 );
+			}
+
+			add_action( 'wp_head', array( $this, 'fix_divi_widget_styles' ) );
+			add_action( 'wp_head', array( $this, 'fix_db_widget_styles' ) );
+		}
+
+		/**
+		 * Remove Elegant Themes' custom posts per page.
+		 *
+		 * Applies to ALL themes by Elegant Themes, not just Divi and Divi-based themes
+		 *
+		 * @see et_custom_posts_per_page()
+		 *
+		 * @param WP_Query $query
+		 */
+		public function remove_et_custom_posts_per_page( $query ) {
+			if ( $query->tribe_is_event_query ) {
+				remove_action( 'pre_get_posts', 'et_custom_posts_per_page' );
+			}
+		}
+
+		/**
+		 * Fix the styling of the Calendar widget and the Advanced List widget
+		 *
+		 * Applies to the Divi theme when Events Calendar PRO is active
+		 */
+		public function fix_divi_widget_styles() {
+
+			// Checking if Events Calendar PRO is active
+			if ( class_exists( 'Tribe__Events__Pro__Main', false ) ) {
```

Comment:
instead of `class_exists`, please do like this: https://github.com/mt-support/tribe-ext-pdf-tickets/blob/1.1.0/tribe-ext-pdf-tickets.php#L104